docs/flatlist.md

```diff
@@ -20,81 +20,37 @@ If you need section support, use [`<SectionList>`](sectionlist.md).
-  renderItem={({item}) => <Text>{item.key}</Text>}
-/>
-```
-
-More complex, multi-select example demonstrating `PureComponent` usage for perf optimization and avoiding bugs.
-
-- By binding the `onPressItem` handler, the props will remain `===` and `PureComponent` will prevent wasteful re-renders unless the actual `id`, `selected`, or `title` props change, even if the components rendered in `MyListItem` did not have such optimizations.
-- By passing `extraData={this.state}` to `FlatList` we make sure `FlatList` itself will re-render when the `state.selected` changes. Without setting this prop, `FlatList` would not know it needs to re-render any items because it is also a `PureComponent` and the prop comparison will not show any changes.
-- `keyExtractor` tells the list to use the `id`s for the react keys instead of the default `key` property.
-
-```javascript
-class MyListItem extends React.PureComponent {
-  _onPress = () => {
-    this.props.onPressItem(this.props.id);
-  };
-
-  render() {
-    const textColor = this.props.selected ? 'red' : 'black';
-    return (
-      <TouchableOpacity onPress={this._onPress}>
-        <View>
-          <Text style={{color: textColor}}>{this.props.title}</Text>
-        </View>
-      </TouchableOpacity>
-    );
-  }
-}
-
-class MultiSelectList extends React.PureComponent {
-  state = {selected: (new Map(): Map<string, boolean>)};
-
-  _keyExtractor = (item, index) => item.id;
-
-  _onPressItem = (id: string) => {
-    // updater functions are preferred for transactional updates
-    this.setState((state) => {
-      // copy the map rather than modifying state.
-      const selected = new Map(state.selected);
-      selected.set(id, !selected.get(id)); // toggle
-      return {selected};
-    });
-  };
-
-  _renderItem = ({item}) => (
-    <MyListItem
-      id={item.id}
-      onPressItem={this._onPressItem}
-      selected={!!this.state.selected.get(item.id)}
-      title={item.title}
-    />
-  );
-
-  render() {
-    return (
+import React from 'react';
+import {StyleSheet, View, FlatList, SafeAreaView, Text} from 'react-native';
+
+const data = [
+  {id: '00', name: 'Ford'},
+  {id: '01', name: 'Tesla'},
+  {id: '02', name: 'BMW'},
+];
+
+export default function App() {
+  return (
+    <SafeAreaView>
       <FlatList
-        data={this.props.data}
-        extraData={this.state}
-        keyExtractor={this._keyExtractor}
-        renderItem={this._renderItem}
+        keyExtractor={(item) => item.id}
+        data={data}
+        renderItem={({item}) => {
+          return (
+            <View>
+              <Text>{item.name}</Text>
+            </View>
+          );
+        }}
       />
-    );
-  }
+    </SafeAreaView>
+  );
 }
 ```
 
-This is a convenience wrapper around [`<VirtualizedList>`](virtualizedlist.md), and thus inherits its props (as well as those of [`<ScrollView>`](scrollview.md)) that aren't explicitly listed here, along with the following caveats:
```

Comment:
This is some vital information that has been removed in the PR. As much as we need the functional component example, we definitely need this section to help users perform complex operations with FlatList